util/dockerfiles/devcontainer/Dockerfile

```
@@ -39,15 +40,60 @@
-RUN git clone --branch stable https://github.com/gem5/gem5
+FROM ghcr.io/gem5/ubuntu-22.04_all-dependencies:latest as builder
+RUN git clone --branch release-staging-v24.0.0.0 https://github.com/gem5/gem5
 WORKDIR /gem5
 RUN scons build/ALL/gem5.opt -j`nproc`
+RUN scons defconfig build/ALL_CHI build_opts/ALL && \
+    scons setconfig build/ALL_CHI RUBY_PROTOCOL_CHI=y && \
+    scons build/ALL_CHI/gem5.opt -j`nproc`
+RUN scons build/VEGA_X86/gem5.opt -j`nproc`
 
 # Stage 2: The final stage where we create the devcontainer image.
 # This includes all dependencies for building and running gem5, the
 # dependencies for developing gem5, and a pre-built gem5 binary.
-FROM --platform=${BUILDPLATFORM} ghcr.io/gem5/ubuntu-24.04_all-dependencies
+FROM ghcr.io/gem5/ubuntu-22.04_all-dependencies:latest
 RUN apt -y update  && \
     apt -y install vim
-COPY --from=builder /gem5/build/ALL/gem5.opt /usr/local/bin/gem5
+    COPY --from=builder /gem5/build/ALL/gem5.opt /usr/local/bin/gem5-default
+    COPY --from=builder /gem5/build/ALL_CHI/gem5.opt /usr/local/bin/gem5
+    COPY --from=builder /gem5/build/VEGA_X86/gem5.opt /usr/local/bin/gem5-vega
+
+# Stage 3: Install rocm stuff
+RUN apt -y update && apt -y upgrade && \
+    apt -y install build-essential git m4 scons zlib1g zlib1g-dev \
+    libprotobuf-dev protobuf-compiler libprotoc-dev libgoogle-perftools-dev \
+    python3-dev python-is-python3 doxygen libboost-all-dev \
+    libhdf5-serial-dev python3-pydot libpng-dev libelf-dev pkg-config
+RUN apt -y install wget
+
+# The follow were adapted from instructions at the URL below, accessed 4/12/24:
+# https://rocm.docs.amd.com/projects/install-on-linux/en/latest/how-to/
+#             native-install/ubuntu.html
+
+# Make the directory if it doesn't exist yet.
+# This location is recommended by the distribution maintainers.
+RUN mkdir --parents --mode=0755 /etc/apt/keyrings
+
+# Download the key, convert the signing-key to a full
+# keyring required by apt and store in the keyring directory
+RUN wget https://repo.radeon.com/rocm/rocm.gpg.key -O - | \
+    gpg --dearmor | tee /etc/apt/keyrings/rocm.gpg > /dev/null
+
+RUN echo "deb [arch=amd64 signed-by=/etc/apt/keyrings/rocm.gpg] https://repo.radeon.com/amdgpu/6.1/ubuntu jammy main" \
+    | tee /etc/apt/sources.list.d/amdgpu.list
+RUN echo "deb [arch=amd64 signed-by=/etc/apt/keyrings/rocm.gpg] https://repo.radeon.com/rocm/apt/6.1 jammy main" \
+    | tee --append /etc/apt/sources.list.d/rocm.list
+# Note: Need to remove 'echo -e' in docker.
+RUN echo 'Package: *\nPin: release o=repo.radeon.com\nPin-Priority: 600' \
+    | tee /etc/apt/preferences.d/rocm-pin-600
+
+RUN apt update
+
+# This package should be the minimum needed to build and is significantly
+# smaller than the full `rocm` package.
+RUN apt -y install rocm-dev
+
+# Stage 4: Install cross compilers
+RUN apt -y install gcc-aarch64-linux-gnu
+RUN apt -y install gcc-riscv64-linux-gnu
```

1. Best to run any `apt` command with the `-y` flag.
2. `apt` `update` and `install` commands should always be in a single `RUN`.
3. Where possible `install` commands should be merged to reduce the Docker image size.

```suggestion
RUN apt -y update && apt -y install \
    # This package should be the minimum needed to build and is significantly
    # smaller than the full `rocm` package.
    rocm-dev \
    # Installs the cross compilers.
    gcc-aarch64-linux-gnu \
    gcc-riscv64-linux-gnu
```